packages/web-components/src/components/rux-tabs/rux-tab/rux-tab.tsx

```diff
@@ -28,13 +37,23 @@ export class RuxTab {
 
     @Element() el!: HTMLRuxTabElement
 
+    /**
+     * Fires when a tab is selected
+     */
+    @Event({ eventName: 'ruxtabselected' }) ruxTabSelected!: EventEmitter
+
     connectedCallback() {
         //handle small on init
         if (this.el?.parentElement?.getAttributeNode('small')) {
             this.el.setAttribute('small', '')
         }
     }
 
+    @Watch('selected')
+    handleSelected() {
+        if (this.selected) this.ruxTabSelected.emit(this.el)
```

Comment:
@markacianfrani I updated this to fire only when selected is changed to be true, otherwise whenever it changes it'll fire this event which doesn't make sense.